jujugui/static/gui/src/app/jujulib/register-user.js

```diff
@@ -0,0 +1,152 @@
+        first parameter and the user data as its second. The user data
+        includes the following fields:
+          - nickname {String} The user's nickname
+          - first {String} The user's first name
+	        - last {String} The user's first name
+	        - email {String} The user's email address
+	        - business {Boolean} whether this is a business account
+          - addresses {Array} A list of address objects (if defined)
+          - vat {String} The VAT number (if defined)
+          - business-name {String} The business name (if defined)
+	        - billing-addresses {Array} A list of billing address objects
+              (if defined)
+	        - payment-methods {Array} A list of payment method objects
+          - allow-email {Boolean} Whether the user allows emails
+          - valid {Boolean} Whether the user is valid
+        If the user is not found, the second argument is null.
+    */
+    getUser: function(name, callback) {
+      const handler = function(error, response) {
+        if (error !== null) {
+          callback(error, null);
+          return;
+        }
+        callback(null, {
+          nickname: response.nickname,
+          first: response.first,
+          last: response.last,
+          email: response.email,
+          business: response.business,
+          addresses: response.addresses,
+          vat: response.vat,
+          businessName: response['business-name'],
+          billingAddresses: response['billing-addresses'],
+          paymentMethods: response['payment-methods'],
+          allowEmail: response['allow-email'],
+          valid: response.valid
+        });
+      };
+      const url = `${this.url}/u/${name}`;
+      return jujulib._makeRequest(this.bakery, url, 'GET', null, handler);
+    },
+
+    /**
+      Add a user.
+
+      @public createUser
+      @param name {String} The user's username.
+      @param user {Object} The user data object, containing:
+        - first {String} The user's first name
+        - last {String} The user's first name
+        - email {String} The user's email address
+        - addresses {Array} A list of address objects (not required)
+        - vat {String} The VAT number (not required)
+        - business {Boolean} whether this is a business account
+        - businessName {String} The business name (not required)
+        - billingAddresses {Array} A list of billing address objects
+            (not required)
+        - allowEmail {Boolean} Whether the user allows emails
+        - token {String} A Stripe token (not required)
+        - paymentMethodName {String} The name of the payment method
+            (not required)
+      @param callback {Function} A callback to handle errors or accept the
+        data from the request. Must accept an error message or null as its
+        first parameter and a user object as its second.
+    */
+    createUser: function(name, user, callback) {
+      const handler = function(error, user) {
```

Comment:
It seems that we have the same handler for both API calls? If so, could this handler be abstracted out?